internal/primitives/state-machine/overlayedchanges/overlayed_changes.go

```diff
@@ -65,3 +73,607 @@ type IndexOperationRenew struct {
 
 func (IndexOperationInsert) isIndexOperation() {}
 func (IndexOperationRenew) isIndexOperation()  {}
+
+type childStorageValue struct {
+	overlayedChangeSet
+	storage.ChildInfo
+}
+
+type OffchainOverlayedChange interface {
+	isOffchainOverlayedChange()
+}
+
+type (
+	OffchainOverlayedChangeRemove   struct{}
+	OffchainOverlayedChangeSetValue []byte
+)
+
+func (OffchainOverlayedChangeRemove) isOffchainOverlayedChange()   {}
+func (OffchainOverlayedChangeSetValue) isOffchainOverlayedChange() {}
```

Comment:
can you move this into `primitives/core`?

Reply:
But I already put this enum in the codebase. https://github.com/ChainSafe/gossamer/blob/diego/overlayed-changes/internal/primitives/core/offchain/offchain.go#L27

Reply:
I think we should keep the `primitives` in the `primitives` folder for now.  After we finish the integration, I think it would make sense to overhaul the `primitives` folder at a later date and move them into `client`, but I think we would still reference the types in the `primitives` folder when encoding/decoding types used in the runtime/FRAME.